packages/plugins/typescript/operations/tests/ts-documents.spec.ts

```diff
@@ -6434,6 +6434,571 @@ function test(q: GetEntityBrandDataQuery): void {
     });
   });
 
+  describe('incremental delivery directive handling', () => {
+    it('should mark fields in deferred fragments as optional (preResolveTypes: true)', async () => {
+      const schema = buildSchema(`
+        type Address {
+          street1: String!
+        }
+
+        type Phone {
+          home: String!
+        }
+
+        type Employment {
+          title: String!
+        }
+
+        type User {
+          name: String!
+          email: String!
+          address: Address!
+          phone: Phone!
+          employment: Employment!
+          widgetCount: Int!
+          widgetPreference: String!
+          clearanceLevel: String!
+          favoriteFood: String!
+          leastFavoriteFood: String!
+        }
+
+        type Query {
+          user: User!
+        }
+      `);
+
+      const fragment = parse(`
+        fragment WidgetFragment on User {
+          widgetCount
+          widgetPreference
+        }
+
+        fragment FoodFragment on User {
+          favoriteFood
+          leastFavoriteFood
+        }
+
+        fragment EmploymentFragment on User {
+          employment {
+            title
+          }
+        }
+
+        query user {
+          user {
+            # Test inline fragment defer
+            ... @defer {
+              email
+            }
+
+            # Test inline fragment defer with nested selection set
+            ... @defer {
+              address {
+                street1
+              }
+            }
+
+            # Test named fragment defer
+            ...WidgetFragment @defer
+
+            # Test a secondary named fragment defer
+            ...FoodFragment @defer
+
+            # Not deferred fields, fragments, selection sets, etc are left alone
+            name
+            phone {
+              home
+            }
+            ...EmploymentFragment
+            ... {
+              clearanceLevel
+            }
+          }
+        }
+      `);
+
+      const { content } = await plugin(
+        schema,
+        [{ location: '', document: fragment }],
+        { preResolveTypes: true },
+        { outputFile: 'graphql.ts' }
+      );
+
+      expect(content).toBeSimilarStringTo(`
+        export type UserQueryVariables = Exact<{ [key: string]: never; }>;
+        export type UserQuery = {
+          __typename?: 'Query',
+          user: {
+            __typename?: 'User',
+            email?: string,
+            clearanceLevel: string,
+            name: string,
+            widgetCount?: number,
+            widgetPreference?: string,
```

Comment:
If I spread `WidgetFragment`, it feels a bit annoying having to do both a null check for `widgetCount` and `widgetPreference`. If one exists the other should exist as well. 🤔

Thus, it might make more sense to implement this with `& { widgetCount?: never; widgetPreference?: never; } | { widgetCoun: number; widgetPreference: string; }` instead.

Reply:
Thanks @n1ru4l, I'm looking into this.